workflow-service/src/main/java/com/redhat/parodos/workflow/definition/parameter/WorkParameterServiceImpl.java

```diff
@@ -33,10 +39,30 @@ public List<WorkParameterValueResponseDTO> getValues(String workflowDefinitionNa
 												.map(workParameterValueRequestDTO -> new ModelMapper().map(
 														workParameterValueRequestDTO, WorkParameterValueRequest.class))
 												.toList())
-								.stream().map(parameterValue -> new ModelMapper().map(parameterValue,
-										WorkParameterValueResponseDTO.class))
-								.toList())
+								.stream()
+								.map(parameterValue -> mappingParameterResponse(workflowDefinitionName, parameterValue))
+								.filter(Objects::nonNull).toList())
 				.orElse(List.of());
 	}
 
+	private WorkParameterValueResponseDTO mappingParameterResponse(String workflowDefinitionName,
+			WorkParameterValueResponse workParameterValueResponse) {
+		String workName = Optional.ofNullable(workParameterValueResponse.getWorkName()).orElse(workflowDefinitionName);
+		Map<String, Object> parameters = workFlowDefinitionService.getWorkParametersByWorkName(workName);
+		if (parameters == null || !parameters.containsKey(workParameterValueResponse.getKey()))
+			return null;
+
+		String propertyPath = workName;
+		WorkFlowDefinition workFlowDefinition = workFlowDefinitionService.getParentWorkFlowByWorkName(workName);
+
+		while (workFlowDefinition != null) {
+			propertyPath = String.join(".", workFlowDefinition.getName(), propertyPath);
+			workFlowDefinition = workFlowDefinitionService.getParentWorkFlowByWorkName(workFlowDefinition.getName());
+		}
+
+		return WorkParameterValueResponseDTO.builder().key(workParameterValueResponse.getKey())
```

Comment:
could we move this to a static method inside the DTO object?

Reply:
yes